File: python/스택큐/같은숫자는싫어.py

""" Level 1. 같은 숫자는 싫어 """

def solution(arr):
    answer = []

    answer.append(arr[0])
    previous = arr[0]

    for num in arr[1:]:

        if previous != num:
            answer.append(num)
            previous = num
        
    return answer


print(solution([1,1,3,3,0,1,1]))        # [1,3,0,1]
print(solution([4,4,4,3,3]))            # [4,3]

File: Java/src/자료구조/같은숫자는싫어.java
package 자료구조;

import java.util.*;

public class 같은숫자는싫어 {

    public static void main(String[] args) {

        int[] a1 = {1,1,3,3,0,1,1};
        System.out.println(Arrays.toString(solution(a1)));

        int[] a2 = {4,4,4,3,3};
        System.out.println(Arrays.toString(solution(a2)));
    }

    public static int[] solution(int []arr) {
        
        List<Integer> answer = new ArrayList<>();
        answer.add(arr[0]);

        int number = arr[0];
        for (int i = 1; i < arr.length; i++) {
            if (number != arr[i]) {
                answer.add(arr[i]);
                number = arr[i];
            }
        }

        return answer.stream()
                .mapToInt(Integer::intValue)
                .toArray();
    }
    
}
